src/js/repeat.js

```diff
@@ -133,40 +133,23 @@ export default {
             return 0;
         }
 
-        let checkEl = el.parentElement.closest( '.or-repeat' );
-        const info = el.classList.contains( 'or-repeat-info' );
-        let count = info ? 1 : Number( el.querySelector( '.repeat-number' ).textContent );
-        const name = el.dataset.name || el.getAttribute( 'name' );
+        const isInfoElement = el.classList.contains( 'or-repeat-info' );
 
+        const toCount = isInfoElement ? `.or-repeat-info[data-name="${el.dataset.name}"]` : `.or-repeat[name="${el.getAttribute( 'name' )}"]`;
+        let count = isInfoElement ? 1 : Number( el.querySelector( '.repeat-number' ).textContent );
```

Comment:
(Not asking for a change here, just making a note.) I was surprised to see a method named `getIndex` which appears to be 1-based (in JavaScript). I see now that [the return value](https://github.com/enketo/enketo-core/pull/782/files#diff-ca036963ec5e89fdaa73c05b2ff5381f3f22d89c8ca700daa74aee65e0821dc8R151) then subtracts to make it 0-based. Also out of scope, but this is another case where I think mixing state and DOM makes code hard to reason about. Of course the user-facing numbers would be 1-based, but that's a representation of state, not *actual* state.